lib/chef/provider/package/chocolatey.rb

```diff
@@ -136,14 +136,27 @@ def check_resource_semantics!
         #
         # @return [String] full path of choco.exe
         def choco_exe
-          @choco_exe ||=
+          @choco_exe ||= begin
+            path = choco_install_path
+
+            # GetEnvironmentVariable returns "" on failure.
+            if path.to_s.empty?
+              raise Chef::Exceptions::MissingLibrary.new("Could not find Chocolatey. Is it installed?")
```

Comment:
I think I like all of the above, e.g., "Could not locate your Chocolatey install. To install chocolatey, we recommend the chocolatey cookbook (URL?). If Chocolatey is installed, ensure that the 'ChocolateyInstall' environment variable is correctly set. You can verify this with the powershell command `[System.Environment]::GetEnvironmentVariable('ChocolateyInstall', 'MACHINE')`".